### PR TITLE
Add a 10 minute delay to surveys

### PR DESCRIPTION
**Pull request by DavidKutu**
For #5261

Also update the threshold and disable times for native and webviews

-   [x] Pull request represents a single change (i.e. not fixing disparate/unrelated things in a single PR).
-   [x] Title summarizes what is changing.
-   [x] Has a [news entry](https://github.com/Microsoft/vscode-jupyter/tree/main/news) file (remember to thank yourself!).
-   [ ] Appropriate comments and documentation strings in the code.
-   [ ] Has sufficient logging.
-   [ ] Has telemetry for enhancements.
-   [ ] Unit tests & system/integration tests are added/updated.
-   [ ] [Test plan](https://github.com/Microsoft/vscode-jupyter/blob/main/.github/test_plan.md) is updated as appropriate.
-   [ ] [`package-lock.json`](https://github.com/Microsoft/vscode-jupyter/blob/main/package-lock.json) has been regenerated by running `npm install` (if dependencies have changed).

### CODE REVIEW

File: src/client/datascience/dataScienceSurveyBanner.ts [hunk 1]
```diff
@@ -56,9 +56,6 @@ enum DSSurveyLabelIndex {
     No
 }
 
-const NotebookOpenThreshold = 5;
-const NotebookExecutionThreshold = 100;
```
Comment (DonJayamanne) [inline src/client/datascience/dataScienceSurveyBanner.ts:60]: Why are sweet removing this?

Comment (DavidKutu) [reply src/client/datascience/dataScienceSurveyBanner.ts:60]: They're on `shouldShowBanner` they now depend on the type of survey

Comment (DavidKutu) [reply src/client/datascience/dataScienceSurveyBanner.ts:60]: here's why:
https://github.com/microsoft/vscode-jupyter/issues/5261#issuecomment-824438141

**Review by rchiodo — APPROVED** [review]

File: src/client/datascience/dataScienceSurveyBanner.ts [hunk 2]
```diff
@@ -169,6 +167,11 @@ export class DataScienceSurveyBanner implements IJupyterExtensionBanner, IExtens
         this.setPersistentState(BannerType.InsidersNotebookSurvey, InsidersNotebookSurveyStateKeys.ShowBanner);
         this.setPersistentState(BannerType.ExperimentNotebookSurvey, ExperimentNotebookSurveyStateKeys.ShowBanner);
         editorProvider.onDidOpenNotebookEditor(this.openedNotebook.bind(this));
+
+        // Change the tenMinutesPassed flag after 10 minutes
+        setTimeout(() => {
```
Comment (IanMatthewHuff) [inline src/client/datascience/dataScienceSurveyBanner.ts:172]: Just a suggestions here. If you ever change this value you have to change it in two spots, and the name of the variable would also be wrong. You could make it a static on the class (so test code could reference it) and call it something like surveyDelay.

**Review by IanMatthewHuff — APPROVED** [review]

**Pull request merged**